components/script/webdriver_handlers.rs

```
@@ -199,6 +226,24 @@ pub unsafe fn jsval_to_webdriver(
                     window.Document().upcast::<Node>().unique_id(),
                 )))
             }
+        } else if object_has_to_json_property(
+            cx,
+            global_scope,
+            jsapi::HandleObject::from_marked_location(&*object),
+        ) {
+            let name = CString::new("toJSON").unwrap();
+            let mut value = UndefinedValue();
+            if JS_CallFunctionName(
+                cx,
+                jsapi::Handle::from_marked_location(&*object),
+                name.as_ptr(),
+                &mut jsapi::HandleValueArray::new(),
+                jsapi::MutableHandle::from_marked_location(&mut value),
+            ) {
+                jsval_to_webdriver(cx, global_scope, Handle::new(&value))
+            } else {
+                Err(WebDriverJSError::UnknownType)
```

We should throw a DOM error here if JS_CallFunctionName failed.